src/jdk.compiler/share/classes/com/sun/tools/javac/comp/TransPatterns.java

```diff
@@ -575,7 +575,8 @@ private void handleSwitch(JCTree tree,
                             }
                         }
 
-                        if (c.guard != null) {
+                        boolean hasGuard = c.guard != null;
```

Comment:
Can this declaration occur before the enclosing `if` ?

Reply:
Extracted the inlined code in a separate step.